KProcessHacker/process.c

```diff
@@ -266,17 +263,12 @@ NTSTATUS KpiOpenProcessToken(
         );
 
     if (!NT_SUCCESS(status))
-    {
-        process = NULL;
-        goto Exit;
-    }
-
-    primaryToken = PsReferencePrimaryToken(process);
+        goto CleanupExit;
 
-    if (!primaryToken)
+    if (!(primaryToken = PsReferencePrimaryToken(process)))
```

Comment:
I really try to avoid assignment in conditionals. It's a risky pattern, there are even are compiler options to disallow them, and my analyzers will mark them as "code smell". That said, if this is a pattern we want to allow - fine to leave this change in.